### ISSUE TITLE
feat: Support single-GPU and low-memory configurations

### ISSUE THREAD

**Issue by newtontech** [Gen-Verse/OpenClaw-RL#11]
## Problem

Current OpenClaw-RL requires **8× GPUs** with high memory, which is a significant barrier for:
- Individual researchers
- Small teams
- Users with limited computational resources

### 1. Gradient Checkpointing
Enable gradient checkpointing to reduce memory usage:
```python
# In config
enable_gradient_checkpointing: true
```

### 2. CPU Offloading
Support offloading model weights to CPU when not in use:
```python
# Memory optimization
cpu_offload: true
offload_ratio: 0.5  # Offload 50% of weights to CPU
```

### 3. Smaller Batch Sizes
Add support for micro-batch accumulation:
```python
# Effective batch size = micro_batch * accumulation_steps
micro_batch_size: 1
gradient_accumulation_steps: 16
```

### 4. Mixed Precision Training
Enable FP16/BF16 training by default:
```python
precision: 'bf16'  # or 'fp16'
```

### 5. Single-GPU Mode
Add a configuration preset for single-GPU training:
```bash
# Quick start for single GPU
export NUM_GPUS=1
export ACTOR_GPUS=1
export ROLLOUT_GPUS=0  # Use actor GPU
export PRM_GPUS=0      # Use actor GPU
bash openclaw-combine/run_single_gpu.sh
```

## Expected Impact

- **Memory**: Reduce from 8×GPU to 1-2×GPU
- **Accessibility**: Enable 10x more users
- **Trade-off**: Slower training speed (acceptable for personal use)

## Acceptance Criteria
- [ ] Single-GPU training works
- [ ] Memory usage < 16GB per GPU
- [ ] Documentation updated
- [ ] Benchmark results provided

## Priority
🟢 High - Increases project accessibility

🦞 OpenClaw AI Assistant
